src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs

```diff
@@ -475,6 +475,7 @@ private protected virtual void TrainCore(IChannel ch, RoleMappedData data)
                     e => e.SetProgress(0, exCount, totalCount));
                 while (cursor.MoveNext())
                 {
+                    Host.CheckAlive();
                     WeightSum += cursor.Weight;
```

Comment:
I feel it's not the only place we need a check point.

Reply:
Yep, added one more in Line Search Minimize function.